src/Foundatio.Repositories.Elasticsearch/Jobs/CleanupIndexesJob.cs

```diff
@@ -19,16 +18,18 @@ namespace Foundatio.Repositories.Elasticsearch.Jobs;
 
 public class CleanupIndexesJob : IJob
 {
-    private readonly IElasticClient _client;
-    private readonly ILogger _logger;
-    private readonly ILockProvider _lockProvider;
+    protected readonly IElasticClient _client;
+    protected readonly ILogger _logger;
+    protected readonly ILockProvider _lockProvider;
+    protected readonly TimeProvider _timeProvider;
     private static readonly CultureInfo _enUS = new("en-US");
     private readonly ICollection<IndexMaxAge> _indexes = new List<IndexMaxAge>();
 
-    public CleanupIndexesJob(IElasticClient client, ILockProvider lockProvider, ILoggerFactory loggerFactory)
+    public CleanupIndexesJob(IElasticClient client, ILockProvider lockProvider, TimeProvider timeProvider, ILoggerFactory loggerFactory)
     {
         _client = client;
         _lockProvider = lockProvider;
+        _timeProvider = timeProvider;
```

Comment:
Should we throw if not set or fall back?

Reply:
Added fallback